packages/auth/src/core/user/user_impl.ts

```diff
@@ -333,4 +333,59 @@ export class UserImpl implements UserInternal {
     await _reloadWithoutSaving(user);
     return user;
   }
+
+  /**
+   * Initialize a User from an idToken server response
+   * @param auth
+   * @param idTokenResponse
+   */
+  static async _fromGetAccountInfoResponse(
+    auth: AuthInternal,
+    response: GetAccountInfoResponse,
+    idToken: string
+  ): Promise<UserInternal> {
+    const coreAccount = response.users[0];
+    _assert(coreAccount.localId !== undefined, AuthErrorCode.INTERNAL_ERROR);
+
+    const providerData: UserInfo[] =
+      coreAccount.providerUserInfo !== undefined
+        ? extractProviderData(coreAccount.providerUserInfo)
+        : [];
+
+    const isAnonymous =
+      !(coreAccount.email && coreAccount.passwordHash) && !providerData?.length;
+
+    const stsTokenManager = new StsTokenManager();
+    stsTokenManager.updateFromIdToken(idToken);
+
+    // Initialize the Firebase Auth user.
+    const user = new UserImpl({
+      uid: coreAccount.localId,
+      auth,
+      stsTokenManager,
+      isAnonymous
+    });
+
+    // update the user with data from the GetAccountInfo response.
+    const updates: Partial<UserInternal> = {
+      uid: coreAccount.localId,
+      displayName: coreAccount.displayName || null,
+      photoURL: coreAccount.photoUrl || null,
+      email: coreAccount.email || null,
+      emailVerified: coreAccount.emailVerified || false,
+      phoneNumber: coreAccount.phoneNumber || null,
+      tenantId: coreAccount.tenantId || null,
+      providerData,
+      metadata: new UserMetadata(
+        coreAccount.createdAt,
+        coreAccount.lastLoginAt
+      ),
+      isAnonymous:
+        !(coreAccount.email && coreAccount.passwordHash) &&
+        !providerData?.length
+    };
+
+    Object.assign(user, updates);
```

Comment:
I didn't know Object.assign modified in place, so that's good to know. In any case, is there a reason to use it instead of spread? I looked it up and the only functional difference is that Object.assign triggers setters which I don't think UserImpl has. I think we usually use spread (can't find a specific style guide rule), so something like `return {...user, ...updates}`.

Reply:
I used Obect.assign because this function is modeled after `_reloadUserWithoutSaving` which does the same thing. Would you still like me to change it?

Reply:
Looks like the `return  {...user, ...updates}' has tsc errors due to the lack of a refresh token in our ServerApp created User. :'/

```
Type '{ displayName: string | null; email: string | null; phoneNumber: string | null; photoURL: string | null; auth: AuthInternal; providerId: ProviderId.FIREBASE; refreshToken?: string | undefined; ... 19 more ...; uid: string; }' is not assignable to type 'UserInternal'.
  Types of property 'refreshToken' are incompatible.
    Type 'string | undefined' is not assignable to type 'string'.
      Type 'undefined' is not assignable to type 'string'.ts(2322)
```